app/src/main/java/eu/kanade/tachiyomi/ui/reader/loader/RarPageLoader.kt

```diff
@@ -58,16 +76,37 @@ internal class RarPageLoader(file: File) : PageLoader() {
 
     override suspend fun getPages(): List<ReaderPage> {
         // SY -->
-        if (readerPreferences.cacheArchiveMangaOnDisk().get()) {
+        if (readerPreferences.archiveReaderMode().get() == ReaderPreferences.ArchiveReaderMode.CACHE_TO_DISK) {
             return DirectoryPageLoader(UniFile.fromFile(tmpDir)!!).getPages()
         }
+        val semaphore = Semaphore(1)
```

Comment:
`Semaphore(1)` is basically a `Mutex()`. I think it should use a Mutex if it needs to avoid multi-threading.